File: ELEC881_Copilot/src/No Description Prompt/Java script (JS)/q23-poor-pigs.js
/**
 * @param {number} buckets
 * @param {number} minutesToDie
 * @param {number} minutesToTest
 * @return {number}
 */
var poorPigs = function(buckets, minutesToDie, minutesToTest) {
    let pigs = 0;
    let tests = Math.floor(minutesToTest / minutesToDie) + 1;
    while (Math.pow(tests, pigs) < buckets) {
        pigs++;
    }
    return pigs;
    
};

File: ELEC881_Tabnine/src/No Description Prompt/Python/q23-poor-pigs.py
class Solution(object):
    def poorPigs(self, buckets, minutesToDie, minutesToTest):
        """
        :type buckets: int
        :type minutesToDie: int
        :type minutesToTest: int
        :rtype: int
        """
        pigs = 0
        while (minutesToTest // minutesToDie) ** pigs < buckets:
            pigs += 1
        return pigs
        